File: python/7 kyu/358.py
# https://www.codewars.com/kata/58d5e6c114286c8594000027/train/python

# Passed

def array_manip(array):
    result = []
    for num_idx, num in enumerate(array):
        right_side_arr = array[num_idx + 1:]
        right_side_filtered_arr = [right_num for right_num in right_side_arr if right_num > num]
        least_greatest_num = min(right_side_filtered_arr) if right_side_filtered_arr else -1
        
        result.append(least_greatest_num)
        
    return result

output = array_manip([8, 58, 71, 18, 31, 32, 63, 92, 43, 3, 91, 93, 25, 80, 28])
print(output)

File: nodejs/7 kyu/358.js
// https://www.codewars.com/kata/580535462e7b330bd300003d/train/javascript

// Passed

function oracle(gestures) {
  let gestureToWinCountMap = new Object();
  let options = ["rock", "paper", "scissors"];

  let optionToPointDetailsMap = {
    rock: {
      rock: 0,
      paper: -1,
      scissors: 1,
    },

    paper: {
      rock: 1,
      paper: 0,
      scissors: -1,
    },
    scissors: {
      rock: -1,
      paper: 1,
      scissors: 0,
    },
  };

  let optionToTotalPoints = new Object();
  for (const option of options) {
    let totalPoints = 0;
    for (const gesture of gestures) {
      let pointsScored = optionToPointDetailsMap[option][gesture];
      totalPoints += pointsScored;
    }

    optionToTotalPoints[option] = totalPoints;
  }

  let totalPointsToOptions = new Object();
  options.map((option) => {
    let totalPoints = optionToTotalPoints[option];
    if (!(totalPoints in totalPointsToOptions)) {
      totalPointsToOptions[totalPoints] = [];
    }

    totalPointsToOptions[totalPoints].push(option);
  });

  let winningOptions = options.filter(
    (option) => optionToTotalPoints[option] > 0,
  );
  let result = winningOptions.length > 0 ? winningOptions.join("/") : "tie";
  return result;
}

const output = oracle(["paper", "scissors", "scissors"]);
console.log(output);